packages/core/src/lib/__tests__/slider.spec.ts

```diff
@@ -0,0 +1,194 @@
+
+    const valuePercentage = (value - customMin) / (customMax - customMin);
+    const style = getComputedStyle(slider);
+    expect(Number.parseFloat(style.left)).toBeCloseTo(valuePercentage * 100);
+  });
+  it('Displays axis ticks at intervals of size step', () => {
+    /*
+     * for reasonable step size, axis ticks display at intervals of size step
+     * does not hold true for relatively small step sizes
+     */
+    const { container } = render(Slider, {
+      min: '-50',
+      max: '50',
+      step: '25',
+    });
+    const axisTicks = container.querySelectorAll('span.bg-gray-6');
+    const numTicks = (customMax - customMin) / step - 1;
+    expect(axisTicks.length).toBe(numTicks);
+    // check positioning of each tick
+    for (const [i, axisTick] of axisTicks.entries()) {
+      const tickPos = getComputedStyle(axisTick).left;
+      expect(Number.parseFloat(tickPos)).toBeCloseTo(
+        ((i + 1) / (numTicks + 1)) * 100,
+        1
+      );
+    }
+  });
+  it('Restricts slider value to intervals of size step', () => {
+    // TODO
+  });
+  it('Renders label above slider', () => {
+    render(Slider, {
+      label: 'sliiide to the left',
+    });
+    const slider = screen.getByText('sliiide to the left', { exact: true });
+    expect(slider).toBeVisible();
+  });
+  it('Renders unit suffix with min and max values', () => {
+    render(Slider, {
+      suffix: 'units',
+      min: '-50',
+      max: '50',
+    });
+    // 1 for min max and one floating value
+    expect(screen.getAllByText('units').length).toBe(3);
+    expect(screen.getByText(customMin, { exact: true })).toBeVisible();
+    expect(screen.getByText(customMax, { exact: true })).toBeVisible();
+  });
+  it('Dispatches "input" event with value when user drags slider to value', async () => {
+    const { container } = render(Slider, {
+      min: '-50',
+      max: '50',
+      value: '15',
+    });
+    const slider = screen.getByRole('slider');
+    expect(slider).toBeVisible();
+
+    // TODO
+  });
+
+  it('Given disabled attributes as true, displays slider as disabled and prevents interaction', () => {
+    // TODO
+  });
+
+  it('Given readonly attributes as true, displays slider as readonly and prevents interaction', () => {
+    // TODO
+  });
+
+  it('Given no attributes, renders slider with { min: 0, max100m value:50, step:1 )', async () => {
+    render(Slider);
+    expect(screen.getByText('0', { exact: true })).toBeVisible();
+    expect(screen.getByText('50', { exact: true }));
+    expect(screen.getByText('100', { exact: true }));
+
+    const slider = screen.getByRole('slider');
+    expect(slider).toBeVisible();
+
+    // TODO Drag isnt working as expected
+    // Drag documentation https://testing-library.com/docs/example-drag/
+    // await fireEvent.mouseDown(slider);
+    // await fireEvent.mouseMove(document, { clientX: -5, y: 0 });
+    // await fireEvent.mouseUp(document);
+    await fireEvent.drag(slider, { delta: { x: -5, y: 0 } });
```

Comment:
@mcous - I was hoping you could give me some guidance on how to best mock dragging the slider for these tests.  Both Kurt and I were having difficulty.  

Note: I'll update the rest of the tests to match the guidelines/comments you gave me in my modal PR.

Reply:
My first reaction would definitely be to use [`userEvent`](https://testing-library.com/docs/user-event/intro/) rather than `fireEvent`; I think I see `userEvent` commented out above. It does a much better job emulating user interactions that trying to `fireEvent` all the different browser events manually

I'd hope something [like this](https://testing-library.com/docs/user-event/pointer#moving-a-pointer) could just work™️